addons/account/models/account_code_mapping.py

```diff
@@ -71,4 +72,16 @@ def _compute_code(self):
 
     def _inverse_code(self):
         for record in self:
-            record.account_id.with_company(record.company_id._origin).code = record.code
+            allowed_company_ids = (record.company_id | self.env.companies).ids
+            record.account_id.with_context({'allowed_company_ids': allowed_company_ids}).code = record.code
+
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
You seem to support both update and create commands in the one2many field, how does it work here?